### PR TITLE
#1115: force users to set slack ids

### LINKED ISSUE #1115: Very Aggressively Force Slack Ids to be set
now they have no choice >:)

### PR DESCRIPTION
**Pull request by anthonybernardi**
## Changes

they have no choice, they can't go to any webpage if it isn't set

i guess they could set a fake one though

## Screenshots

Screenshot 2023-04-24 at 10 41 59 AM

Closes #1115

### CODE REVIEW

File: src/frontend/src/pages/HomePage/Home.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const Home = () => {
   const history = useHistory();
   const { isLoading, isError, error, data: userSettingsData } = useSingleUserSettings(user.userId);
 
-  if (isLoading) return <LoadingIndicator />;
+  if (isLoading || !userSettingsData) return <LoadingIndicator />;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/Home.tsx:21]: lmao they literally cant use it no matter what

**Review by Peyton-McKee — APPROVED** [review]
looks good if the loading thing makes sense

File: src/frontend/src/app/AppPublic.tsx [hunk 2]
```diff
@@ -23,14 +23,14 @@ const AppPublic: React.FC = () => {
...
-      return <AppAuthenticated />;
+      return <AppAuthenticated userId={auth.user.userId} />;
     }
 
     // if we're on development and the userId is stored in localStorage,
     // then dev login right away (no login page redirect needed!)
     if (process.env.NODE_ENV === 'development' && devUserId) {
       auth.devSignin(parseInt(devUserId));
-      return auth.isLoading ? <LoadingIndicator /> : <AppAuthenticated />;
+      return <LoadingIndicator />;
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/app/AppPublic.tsx:33]: so the way this was before didn't even make sense -- if we got to this if statement, it means that auth did not exist yet. once auth does exist, this render function gets recalled so it goes up to the top block and this whole `: <AppAuthenticated />` was totally pointless

very weird concept, i wrote this originally and didn't realize it until now

File: src/frontend/src/pages/HomePage/Home.tsx (hunk 1 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/HomePage/Home.tsx:21]: lol yeah this block will never occur anymore but whatever

**Review by RChandler234 — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**